src/main/msp/msp.c

```diff
@@ -3254,13 +3254,13 @@ static mspResult_e mspProcessInCommand(mspDescriptor_t srcDesc, int16_t cmdMSP,
         }
         if (sbufBytesRemaining(src) >= 6) {
             // Added in MSP API 1.40
-            rxConfigMutable()->rcInterpolationChannels = sbufReadU8(src);
+            sbufReadU8(src); // not required in API 1.44, was rxConfigMutable()->rcSmoothingChannels
 #if defined(USE_RC_SMOOTHING_FILTER)
-            configRebootUpdateCheckU8(&rxConfigMutable()->rc_smoothing_type, sbufReadU8(src));
-            configRebootUpdateCheckU8(&rxConfigMutable()->rc_smoothing_input_cutoff, sbufReadU8(src));
-            configRebootUpdateCheckU8(&rxConfigMutable()->rc_smoothing_derivative_cutoff, sbufReadU8(src));
-            configRebootUpdateCheckU8(&rxConfigMutable()->rc_smoothing_input_type, sbufReadU8(src));
-            configRebootUpdateCheckU8(&rxConfigMutable()->rc_smoothing_derivative_type, sbufReadU8(src));
+            configRebootUpdateCheckU8(&rxConfigMutable()->rc_smoothing_mode, sbufReadU8(src));
```

Comment:
I am concerned about the re-use of this byte - `rc_smoothing_type` is defined differently from `rc_smoothing_mode`, so in order to not risk getting problematic configurations from legacy applications, we should be discarding this byte, and add one for `rc_smoothing_mode` at the end  of the frame.

Reply:
I've thought about this at length and thing the risk is exceedingly low if we leave it as is. I say this because:

use of the old linear interpolation smoothing mode mode is vanishingly rare
it hasn't been default for some years now, and no-one has every encouraged users to go back or try the old method
the user name has changed in CLI, so copy and paste updaters who used linear interpolation will get safe filters
only people who upgraded without erasing flash and were using linear interpolation would have a problem. Not having RC smoothing isn't fatal; it would be the least of their problems.
Hence I personally feel we can leave it as is.

Reply:
This isn't about the CLI, or upgrading. This is about MSP, and the way backwards compatibility is achieved in it. And the way is that no field in a message is ever re-used - if a parameter is removed, the field for it is ignored, and 0 returned in queries. New parameters are appended at the end.
This is to make sure that users do not get unexpected configurations when they are using any legacy apps / OSD devices / lua scripts that communicate with MSP and do not 'know' about the change.
If a new parameter is appended at the end, these apps will just ignore it, and keep working as intended. If a parameter is replaced, the result will be a flight controller configuration that does not match what the app shows to the user. So this is not about risk, but about going for the approach that avoids breaking apps without need for it.

Reply:
Really sorry that I got confused with the CLI.  Have attempted an map fix.